src/messages/manifestjson.js

```diff
@@ -107,6 +124,27 @@ export const MANIFEST_PERMISSIONS = {
   file: MANIFEST_JSON,
 };
 
+export const MANIFEST_PERMISSIONS_PRIVILEGED =
+  'MANIFEST_PERMISSIONS_PRIVILEGED';
+export function manifestPermissionsPrivileged(error) {
+  const messageTmpl =
+    i18n._(oneLine`%(instancePath)s: privileged permissions (%(privilegedPermissions)s)
+                     are only allowed in add-ons signed as privileged.`);
+  const message = i18n.sprintf(messageTmpl, {
+    instancePath: error.instancePath,
+    privilegedPermissions: error.params.privilegedPermissions
+      .map((p) => `"${p}"`)
+      .join(', '),
```

Comment:
nit: I think we should add this list at the end and use `JSON.stringify(error.params.privilegedPermissions)` so that it outputs something like this:

```
/instancePath: the following privileged permissions are only allowed in add-ons signed as privileged: ["foo", "bar"]
```